src/vs/platform/theme/browser/iconsStyleSheet.ts

```diff
@@ -31,34 +31,41 @@ export function getIconsStyleSheet(themeService: IThemeService | undefined): IIc
 		getCSS() {
 			const productIconTheme = themeService ? themeService.getProductIconTheme() : new UnthemedProductIconTheme();
 			const usedFontIds: { [id: string]: IconFontDefinition } = {};
-			const formatIconRule = (contribution: IconContribution): string | undefined => {
+
+			const rules: string[] = [];
+			const rootAttribs: string[] = [];
+			for (const contribution of iconRegistry.getIcons()) {
 				const definition = productIconTheme.getIcon(contribution);
 				if (!definition) {
-					return undefined;
+					continue;
 				}
+
 				const fontContribution = definition.font;
+				const fontFamilyVar = `--vscode-icon-${contribution.id}-font-family`;
+				const contentVar = `--vscode-icon-${contribution.id}-content`;
 				if (fontContribution) {
 					usedFontIds[fontContribution.id] = fontContribution.definition;
-					return `.codicon-${contribution.id}:before { content: '${definition.fontCharacter}'; font-family: ${asCSSPropertyValue(fontContribution.id)}; }`;
-				}
-				// default font (codicon)
-				return `.codicon-${contribution.id}:before { content: '${definition.fontCharacter}'; }`;
-			};
-
-			const rules = [];
-			for (const contribution of iconRegistry.getIcons()) {
-				const rule = formatIconRule(contribution);
-				if (rule) {
-					rules.push(rule);
+					rootAttribs.push(
+						`${fontFamilyVar}: ${asCSSPropertyValue(fontContribution.id)};`,
+						`${contentVar}: '${definition.fontCharacter}';`,
+					);
+					rules.push(`.codicon-${contribution.id}:before { content: '${definition.fontCharacter}'; font-family: ${asCSSPropertyValue(fontContribution.id)}; }`);
+				} else {
+					rootAttribs.push(`${contentVar}: '${definition.fontCharacter}';`);
```

Comment:
We should set `${fontFamilyVar}: 'codicon'` and not rely on the undefined variable behavior.